source/CAS/Client.php

```diff
@@ -1570,50 +1722,50 @@ private function _wasPreviouslyAuthenticated()
         if ( $this->isProxy() ) {
             // CAS proxy: username and PGT must be present
             if ( $this->isSessionAuthenticated()
-                && !empty($_SESSION['phpCAS']['pgt'])
+                && $this->hasSessionValue('pgt')
             ) {
                 // authentication already done
-                $this->_setUser($_SESSION['phpCAS']['user']);
-                if (isset($_SESSION['phpCAS']['attributes'])) {
-                    $this->setAttributes($_SESSION['phpCAS']['attributes']);
+                $this->_setUser($this->getSessionValue('user'));
+                if ($this->hasSessionValue('attributes')) {
+                    $this->setAttributes($this->getSessionValue('attributes'));
                 }
-                $this->_setPGT($_SESSION['phpCAS']['pgt']);
+                $this->_setPGT($this->getSessionValue('pgt'));
                 phpCAS::trace(
-                    'user = `'.$_SESSION['phpCAS']['user'].'\', PGT = `'
-                    .$_SESSION['phpCAS']['pgt'].'\''
+                    'user = `'.$this->getSessionValue('user').'\', PGT = `'
+                    .$this->getSessionValue('pgt').'\''
                 );
 
                 // Include the list of proxies
-                if (isset($_SESSION['phpCAS']['proxies'])) {
-                    $this->_setProxies($_SESSION['phpCAS']['proxies']);
+                if ($this->hasSessionValue('proxies')) {
```

Comment:
Currently the equivalent in `hasSessionValue` is `isset && !empty` which is not appropriate here (`_setProxies` needs to override `$_proxies` as needed)